dataprofiler/profilers/profile_builder.py

```diff
@@ -683,3 +687,127 @@ def tqdm(l):
+        Helper method for restoring all data labelers after saving to or 
+        loading from disk.
+        """
+        # Restore data labeler for profiler
+        data_labeler_options = self.options.structured_options.data_labeler
+        if data_labeler_options.is_enabled \
+                and data_labeler_options.data_labeler_object is None:
+            try:
+                data_labeler = DataLabeler(
+                    labeler_type='structured',
+                    dirpath=data_labeler_options.data_labeler_dirpath,
+                    load_options=None)
+                self.options.set({'data_labeler.data_labeler_object': data_labeler})
+                
+            except Exception as e:
+                utils.warn_on_profile('data_labeler', e)
+                self.options.set({'data_labeler.is_enabled': False})
+                
+        # Restore data labelers for all columns
+        for key in self._profile:
+
+            val = self._profile[key]
+            use_data_labeler = True
+            if val.options and isinstance(val.options, StructuredOptions):
+                use_data_labeler = val.options.data_labeler.is_enabled
+
+            if use_data_labeler:
+                data_labeler_profile = val.profiles['data_label_profile']._profiles['data_labeler']
+                data_labeler_profile.data_labeler = None
+                
+                if val.options and val.options.data_labeler.data_labeler_object:
+                    data_labeler_profile.data_labeler = val.options.data_labeler.data_labeler_object
+                if data_labeler_profile.data_labeler is None:
+                    data_labeler_dirpath = None
+                    if val.options:
+                        data_labeler_dirpath = val.options.data_labeler.data_labeler_dirpath
+                
+                    data_labeler_profile.data_labeler = DataLabeler(
+                        labeler_type='structured',
+                        dirpath=data_labeler_dirpath,
+                        load_options=None)
+
+
+    def save(self, filepath="profile.pkl"):
+        """
+        Save profiler to disk
+        
+        :param filepath: Path of file to save to
+        :type filepath: String
+        :return: None
+        """
+        # Delete data labelers as they can't be pickled
+        self._delete_data_labelers()
+
+        # Create dictionary for all metadata, options, and profile 
+        data = { 
+                "total_samples": self.total_samples,
+                "encoding": self.encoding,
+                "file_type": self.file_type,
+                "row_has_null_count": self.row_has_null_count,
+                "row_is_null_count": self.row_is_null_count,
+                "hashed_row_dict": self.hashed_row_dict,
+                "_samples_per_update": self._samples_per_update,
+                "_min_true_samples": self._min_true_samples,
+                "options": self.options,
+                "_profile": self.profile
```

Comment:
The options classes and numpy arrays are two of obstacles in saving the entire profile as a JSON. Any recommendations on how I might be able to save (and later load) these two objects to JSON?

Reply:
Only option for saving as a JSON is really writing them as lists or dicts to the profile.

Reply:
Options at some point should get serializable capability would be extremely helpful